STYLE_GUIDE.md

```diff
@@ -0,0 +1,190 @@
+# Go Coding Style Guide
+
+In order to keep our code looking good with lots of programmers working on it, it helps to have a "style guide", so all
+the code generally looks quite similar. This doesn't mean there is only one "right way" to write code, or even that this
+standard is better than your style.  But if we agree to a number of stylistic practices, it makes it much easier to read
+and modify new code. Please feel free to make suggestions if there's something you would like to add or modify.
+
+Please, before going through the rest of the document, have a read over the [syntactical style](https://github.com/uber-go/guide/blob/master/style.md)
+
+If you wish to gain some further understanding, we recommend having a read on how to code [effective go](https://golang.org/doc/effective_go.html)
+
+
+
+## Code Structure
+
+Perhaps more key for code readability than good commenting is having the right structure. As a rule of thumb, try to write
+in a logical order of importance, taking a little time to think how to order and divide the code such that someone could
+scroll down and understand the functionality of it just as well as you do. A loose example of such order would be:
+* Main Struct
+* Options (only if they are seen as critical to the struct else they should be placed in another file)
+* Initialisation / Start and stop of the service
+* Msgs/Events
+* Public Functions (In order of most important)
+* Private/helper functions
+* Auxiliary structs and function (can also be above private functions or in a separate file)
+
+## General
+
+ * Use `gofmt` (or `goimport`) to format all code upon saving it.  (If you use VIM, check out vim-go).
+ * Use a linter (see below) and generally try to keep the linter happy (where it makes sense).
+ * Think about documentation, and try to leave godoc comments, when it will help new developers.
+ * Every package should have a high level doc.go file to describe the purpose of that package, its main functions, and any other relevant information.
+ * `TODO` should not be used in lieu of correctness.  It can be used sparingly to document future direction.
+ * `BUG` / `FIXME` should be avoided; these should instead be documented as specific issues that can be linked to in the code.
```

Comment:
Really? I find `FIXME`s are usually minor issues that will sit around until the end of time, and are usually left as markers for gotchas that a future developer needs to be aware of when working with the code. Putting them in an issue will often just make them non-discoverable from the code, and clog up the issue backlog.

Reply:
Would you like to remove it then? I would have thought that if it was small enough that there be no reason that whoever wrote it or noticed it would be able to solve it then and there and if it was no small fix, that an issue would be appropriate

Reply:
Personally I quite like `FIXME`s, but maybe we could consider saying that every `FIXME` should refer to an issue?